### DISCUSSION TITLE
Flow Economy - an Alternative Optional Economy System

### DISCUSSION THREAD

**Discussion by marcushutchings** [beyond-all-reason/rfcs#6] [Ideas]
Discussion to support https://github.com/beyond-all-reason/rfcs/pull/5

**Comment by Beherith** [comment]
I likey. Full speed ahead!

**Comment by WatchTheFort** [comment]
> Would be able to drop the need for economy prioritization widgets.

I am 1,000% in favour of this

**Comment by sprunk** [comment]
> Would be able to drop the need for economy prioritization widgets.

No, why? It only lets game devs simplify economy prioritisation gadgets. Most types of prioritisation widgets will stay because the technicalities behind the resource system do not broadly impact player decisions.

For example:
 * ZK and BAR has a gadget that makes sure factories at high priority get resource flow before factories at low priority. This will stay, because something still has to tell the new system which units are high priority and which aren't. It should become possible to simplify this type of gadget though, but not drop it completely.
 * somebody may write a gadget that makes sure two factories building the same unit will, over time, have spent the same amount of metal. I don't know about BAR, but ZK tries to implement a fair proration but as far as I can tell in pathological cases it can still fail since it still relies on the engine system. If the engine system prorates correctly then ZK will be able to largely drop that part of the gadget, though not entirely because proration is only equalized across the same priority class, and priorities stay (see the bullet point above).
 * players may have broader, strategic considerations of how resources should flow. For example if they start a superweapon, they may want to drop other projects. This type of widget is a higher-level that interfaces with the resource system on a decision-making level (for example by setting low priority, or stopping production altogether), these generally don't mind small inaccuracy in resource flow. I don't think I've ever seen a widget deal with low-level resourcing issues directly.

That said, it would still be good to have, just not essential.
